### PR TITLE
Take wireserver address from option 245 in DHCP lease

### LINKED ISSUE #1176: don't use the dns server from dhcp as the api endpoint
https://github.com/Azure/WALinuxAgent/blob/master/azurelinuxagent/common/osutil/default.py#L846

get_endpoint_from_leases_path incorrectly uses the HEADER_DNS as the API endpoint when instead the endpoint should be pulled from dhcp option 245 using get_dhcp_lease_endpoint (which isn't implemented yet) 

This breaks an Azure deployment if an end user modifies a vnet to use a custom DNS server because waagent no longer uses the correct KNOWN_WIRESERVER_IP or option 245 from the DHCP lease file.

### PR DESCRIPTION
**Pull request by narrieta**
## Description

Fixes #1176

On default vnet configurations, the wireserver address shows up both on domain-name-servers and option 245. Currently we take the address from domain-name-servers. If the vnet has a custom DNS server, domain-name-servers will have the address of that server, not of wireserver. Changed the code to take the address from option 245 instead.

---

### PR information
- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/data/dhcp.leases.custom.dns [hunk 1]
```diff
@@ -0,0 +1,56 @@
+lease {
```
Comment (boumenot) [inline tests/data/dhcp.leases.custom.dns]: Is this a typical example?  

The same lease is repeated three times. Is this is expected. When you are searching for `option unknown-245` should you look for the first or last? Does it matter?  Should you take the interface into account, e.g. interface "eth0" when parsing?

File: azurelinuxagent/common/osutil/default.py [hunk 2]
```diff
@@ -937,28 +937,23 @@ def get_endpoint_from_leases_path(pathglob):
         endpoint = None
 
         HEADER_LEASE = "lease"
-        HEADER_OPTION = "option unknown-245"
-        HEADER_DNS = "option domain-name-servers"
+        HEADER_OPTION_245 = "option unknown-245"
```
Comment (boumenot) [inline azurelinuxagent/common/osutil/default.py]: OpenBSD referecnes HEADER_OPTION too.  Do any changes need to be made there?

Comment (narrieta) [reply azurelinuxagent/common/osutil/default.py]: Yup, good catch. Will update there as well.

File: tests/data/dhcp.leases.custom.dns (hunk 1 above)
Comment (narrieta) [reply tests/data/dhcp.leases.custom.dns]: This is the same file as used for the other unit tests (except for the last entry, which has a custom DNS server: 8.8.8.8). I would say it is typical, looks very similar to the one on my VM. According to man, the file acts as a log and the last entry should always be used.

Actually, I need to update the address in option 245 to a dummy address, to ensure the last one is being picked (with the previous code this was ensured by using 'invalid' and 'expired' in domain-name-servers of the first 2 entries.

Comment (narrieta) [reply tests/data/dhcp.leases.custom.dns]: I just updated the test file -- added dummy wireserver addresses to the first 2 entries to ensure the test is pickling up the last one (third one).

File: azurelinuxagent/common/osutil/default.py (hunk 2 above)
Comment (narrieta) [reply azurelinuxagent/common/osutil/default.py]: After further discussion I will leave OpenBSD as is for now. Thanks for pointing this out.

**Pull request merged**